Mathlib/CategoryTheory/Comma/Over.lean

```diff
@@ -657,4 +657,114 @@ lemma toUnder_comp_forget (F : S ⥤ T) (X : T) (f : (Y : S) → X ⟶ F.obj Y)
 
 end Functor
 
+namespace StructuredArrow
+
+/-- Characterization of the structured arrow category on the projection functor of any
+structured arrow category. -/
+def ofStructuredArrowProjEquivalence {D : Type u₂} [Category.{v₂} D] (F : D ⥤ T) (Y : T) (X : D) :
+    StructuredArrow X (StructuredArrow.proj Y F) ≌ StructuredArrow Y (Under.forget X ⋙ F) :=
+  Equivalence.mk
+    (Functor.toStructuredArrow
+      (Functor.toUnder (StructuredArrow.proj X _ ⋙ StructuredArrow.proj Y _) _
+        (fun g => by exact g.hom) (fun m => by have := m.w; aesop_cat)) _ _
+      (fun f => f.right.hom) (by simp))
+    (Functor.toStructuredArrow
+      (Functor.toStructuredArrow (StructuredArrow.proj Y _ ⋙ Under.forget X) _ _
+         (fun g => by exact g.hom) (fun m => by have := m.w; aesop_cat)) _ _
+      (fun f => f.right.hom) (by simp))
+    (NatIso.ofComponents (fun x => by
+      simp only [Functor.id_obj, Functor.const_obj_obj, proj_obj, left_eq_id, Functor.const_obj_map,
```

Comment:
Wait, isn't this still generating data using tactics? Can the `simp only` be replaced by `dsimp only` at least?

Reply:
You're right.... I tried that to make it faster than just to use `Iso.refl _`... Maybe I should try to piece the iso together instead...